reactioncommerce/reaction-component-library
# feat: AddressForm Component

Resolves #121 
Impact: **minor**  
Type: **feature**





## Component
Create an AddressForm component using [reacto forms](http://forms.dairystatedesigns.com/reacto-form/) and preexisting form components

**Validation**
The Address Form component does a simple `isRequired` validation on all required fields. reacto-forms let you provide your own form validation as long as it's based on this [spec](http://forms.dairystatedesigns.com/spec/validation/) this will allow users to add custom validation during implementation.

**PhoneNumberInput**
Included in this PR is a very MVP of a PhoneNumberInput component. Currently, it's just a wrapper around the TextInput that trims all non-digit chars from the entered string. 
*Example:* (504) 393-7303 -> 5043937303

In the future, we'd like to have a more feature rich PhoneNumberInput that includes masking and real phone number validation.
*Examples:* 
http://catamphetamine.github.io/react-phone-number-input/
https://nosir.github.io/cleave.js/

## Screenshots
Empty
<img width="514" alt="screen shot 2018-07-16 at 10 50 45 am" src="https://user-images.githubusercontent.com/1135948/42768880-3f373bfe-88e6-11e8-8577-cf950fd2c983.png">

Editing
<img width="504" alt="screen shot 2018-07-16 at 10 50 54 am" src="https://user-images.githubusercontent.com/1135948/42768891-473c3dae-88e6-11e8-9448-8fe91e0032d6.png">

Mobile
<img width="418" alt="screen shot 2018-07-16 at 10 51 11 am" src="https://user-images.githubusercontent.com/1135948/42768901-4e63513a-88e6-11e8-805d-6ef9d07a2d9a.png">

## Breaking changes
N/A

## Notes
@cassytaylor  & @rymorgan  since this component was built inside the component library I've used the input components we built several months ago so they're based on [these](https://zpl.io/bW1Xg61) field designs

## Testing
1. Verify you can submit and edit an address
2. Verify the new `PhoneNumberInput` will trim non-digit characters and only returns a formatted phone number (i.e: 504-123-1234 => 5041231234)
